### ISSUE TITLE
Returning calls in 1000 genomes is slow

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#950]
When there is a VCF with many callsets in it, responses can become slow.

**Comment by jeromekelleher** [comment]
Need some more details here @david4096... what's your pagesize in particular? Are you sure it's not the client side JSON decoding that's the bottleneck? We have some perf benchmarks for this sort of thing in ``server_benchmark.py`` if you want to have a look. They might need to be updated a bit though; see #711.

**Comment by kozbo** [comment]
I was running some tests against the 1000 Genomes VCF files (2504 samples) and I verified the slow results. I ran some tests with different page sizes and the amount of time to return each page increased linearly with the number of variants included in each page. I also tried changing my search variants to use `callSetIds=[]` but that made no difference hoping that I could avoid the callSetId searching in convertVariant, but that didn't work either because we don't check for the empty callset. So then we did some profiling and we found that we were spending a lot of time in base64 (which is probably why the CPU was so busy in top). Once we passed in the empty array and also modified the code in variants.py to not search if the length of the callSetId array is 0, and then my search ran very quickly. All this will see explained in @david4096 's PR coming soon.

We learned a lot about profiling the server so I thought I would capture the tricks here to start.
No surprise, you cannot easily profile from the Apache environment. Also we turned off the flask debugger and we also had to prevent the use of the reloader. Here is the resulting command line we used:
`python -m cProfile -s cumulative -o profile.out ko-server/server_dev.py -H 100.70.26.73 --dont-use-reloader --config-file /srv/ga4gh/config.py`

We also found a neat way to investigate the output of the profiler via a browser.
`pip install cprofilev
cprofilev -f profile.out -a 100.70.26.73`
